src/study/system-model/component.cpp

```diff
@@ -118,9 +118,11 @@ ComponentBuilder& ComponentBuilder::withScenarioGroupId(const std::string& scena
  *
  * \return The constructed Component object.
  */
-Component ComponentBuilder::build() const
+Component ComponentBuilder::build()
 {
-    return Component(data_);
+    Component component(data_);
+    data_.reset(); // makes the ComponentBuilder re-usable
+    return component;
```

Comment:
naah

Reply:
I don't understand

Reply:
Why reset data_ every time? If the data is specific to each component, it should be passed directly as an argument to the build method, eliminating the need for a reset method and the data_ member altogether.

If the goal is to create a large number of components that may share some common data, the build method could include an optional boolean parameter to selectively reset all or part of the builder's state, providing flexibility where needed.